### PR TITLE
update mypy strict list

### PR DESCRIPTION
**Pull request by altendky**
Draft for:
- [x] Merge this when other important PRs aren't waiting
- [x] Fixup the newly identified type hint issues

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -7,7 +7,7 @@ warn_unused_ignores = True
 [mypy - lib]
 ignore_errors = True
 
-[mypy-chia.data_layer.*,tests.core.data_layer.*,chia.cmds.data,chia.cmds.data_funcs,chia.rpc.data_layer_rpc_api,chia.rpc.data_layer_rpc_client,chia.server.start_data_layer,chia.wallet.db_wallet.*,chia.wallet.dlo_wallet.*,chia.wallet.util.merkle_tree,tests.wallet.db_wallet.*]
+[mypy-chia.cmds.data,chia.cmds.data_funcs,chia.data_layer.*,chia.rpc.data_layer_rpc_api,chia.rpc.data_layer_rpc_client,chia.server.start_data_layer,chia.wallet.db_wallet.*,chia.wallet.dlo_wallet.*,chia.wallet.util.merkle_tree,tests.core.data_layer.*,tests.wallet.db_wallet.*,tests.wallet.rpc.test_dl_wallet_rpc]
```
Comment (altendky) [inline mypy.ini:10]: A reorder plus adding `tests.wallet.rpc.test_dl_wallet_rpc` to the stricter list.  Note that recently on `main` strict has become the default so this list will get deleted when that is merged into atari.

File: chia/rpc/rpc_server.py [hunk 2]
```diff
@@ -19,6 +20,11 @@
 log = logging.getLogger(__name__)
 
 
+class RpcApiProtocol(Protocol):
+    def get_routes(self) -> Dict[str, Callable[[Any], Any]]:
+        pass
```
Comment (altendky) [inline chia/rpc/rpc_server.py:25]: I expect there will be more use cases for a more filled out version of this when we fill out more hints in more files.

File: tests/wallet/rpc/test_dl_wallet_rpc.py [hunk 3]
```diff
@@ -8,20 +10,27 @@
...
+from chia.wallet.wallet_node import WalletNode
+
 from tests.setup_nodes import bt, setup_simulators_and_wallets, self_hostname
 from tests.time_out_assert import time_out_assert
 from tests.util.rpc import validate_get_routes
 
 log = logging.getLogger(__name__)
 
 
+SimulatorsAndWallets = Tuple[List[FullNodeSimulator], List[Tuple[WalletNode, ChiaServer]]]
```
Comment (altendky) [inline tests/wallet/rpc/test_dl_wallet_rpc.py:28]: This absolutely belongs elsewhere and for all.  This is a stop gap here until we get to that on `main`.

File: tests/wallet/rpc/test_dl_wallet_rpc.py [hunk 4]
```diff
@@ -144,9 +154,11 @@ async def is_singleton_confirmed(rpc_client, lid) -> bool:
 
             await client_2.dl_track_new(launcher_id)
 
-            async def is_singleton_generation(rpc_client, lid, generation) -> bool:
+            async def is_singleton_generation(rpc_client: WalletRpcClient, lid: bytes32, generation: int) -> bool:
                 if await is_singleton_confirmed(rpc_client, lid):
                     rec = await rpc_client.dl_latest_singleton(lid)
+                    if rec is None:
+                        raise Exception("No latest singleton for: {lid!r}")
```
Comment (altendky) [inline tests/wallet/rpc/test_dl_wallet_rpc.py:161]: The next line would have raised an `AttributeError` so at the level of 'an exception is raised' there is no change here.

**Review by altendky — COMMENTED** [review]
I would like to land this before we catch up merge from `main` into `atari`.

**Review by Quexington — APPROVED** [review]
Not sure I'm the person to review all of this, but the wallet test changes look accurate.

**Pull request merged**